packages/graphback-core/src/db/buildModelTableMap.ts

```diff
@@ -0,0 +1,68 @@
+import { GraphQLField, GraphQLFieldMap, GraphQLObjectType } from "graphql";
+import { parseDbAnnotations } from '../annotations/parser';
+import { defaultColumnNameTransform, defaultTableNameTransform } from './defaultNameTransforms';
+import { getPrimaryKey } from './getPrimaryKey';
+
+/**
+ * Contains mapping information between GraphQL Model type and database table
+ * 
+ * - typeName: Original GraphQLObjectType name
+ * - tableName: Name of datase table
+ * - id: Indicates the primary key field
+ * - fieldMap: Object of key-value mapping between GraphQL fields and database columns.
+ */
+export interface ModelTableMap {
+  typeName: string
+  tableName: string
+  idField: string
+  fieldMap?: any
+}
+
+export const buildModelTableMap = (model: GraphQLObjectType): ModelTableMap => {
+  const primaryKeyField = getPrimaryKey(model);
+  const tableName = getTableName(model);
+  const fieldMap = mapFieldsToColumns(model.getFields());
+
+  return {
+    idField: primaryKeyField ? primaryKeyField.name : undefined,
+    typeName: model.name,
+    tableName,
+    fieldMap
+  }
+}
+
+function mapFieldsToColumns(fieldMap: GraphQLFieldMap<any, any>): any {
+  return Object.values(fieldMap).reduce((obj: any, field: GraphQLField<any, any>) => {
+    const columnName = getColumnName(field);
+
+    if (field.name !== columnName) {
+      obj[field.name] = columnName;
+    }
+
+    // TODO: Map relationship fields
+
+    return obj;
+  }, {});
+}
+
+export function getTableName(model: GraphQLObjectType): string {
+  let mappedTableName = defaultTableNameTransform(model.name, 'to-db');
+
+  const dbAnnotations = parseDbAnnotations(model);
```

Comment:
I think that will be really hard to document and very confusing. 
We are basically going to say that DB migrations are optional and only to be used when
we do have some db, but then people will need to use some subset of the db annotations to do mapping (even when their data source is REST etc.)

Not sure what we can do with this and if that is doable in current PR, but raising as possible issue

Reply:
I don't see how it is confusing? database migrations are optional, but the `@db.name` annotation is simply a way to map to your existing column.

Reply:
The other option is to introspect an existing database and generated a schema from that.

Reply:
Agreement made to remove default transformations and let people bring their own with annotations only.